### DISCUSSION TITLE
get specific translation in api

### DISCUSSION THREAD

**Discussion by pandora-otto** [directus/directus#6955] [Archive]
hi guys

First, I'd like to congratulate the team for the creation of Directus, I'm very new user of this beautiful software and I'm looking forward for the next updates.

I have a question though regarding translations.
I have created a collection 'Articles' with a 'name' and 'translations' fields. In 'Article Translations' I have created two fields, 'title' and 'text'.

My question is how do I use the rest api to get articles or an article for a specific language?
I have tried the following but do not seem to work.

1. `/items/articles?deep={"translations":{"_filter":{"languages_code":{"_eq":"de-DE"}}}}`
2. `/items/articles?filter={"language_code":{"_eq":"de-DE"}}`
3. `/items/articles?translations{filter={"language_code":{"_eq":"de-DE"}}}`
4. `/items/articles?fields[]=name&fields[]=translations{filter={"language_code":{"_eq":"de-DE"}}}`
5. `/items/articles?fields[]=name&fields[]=translations&filter={"language_code":{"_eq":"de-DE"}}`

Is it achievable what I'm tried to do here or it's something that will be able to be done at later updates?

Any advice is welcome.

many thanks
Otto

**Comment by rijkvanzanten** [comment]
Number 1 should be correct, though you have a typo -> "language **s** _code" I believe that should be `language_code` if you're using the default field names 👍🏻

**Reply by pandora-otto** [reply]
thnx so much @rijkvanzanten 

I do use the default field name. I've tried number 1 again with your suggestion.
1a. `/items/articles?deep={"translations":{"_filter":{"language_code":{"_eq":"de-DE"}}}}`

but the response is not nice :)
`{"errors":[{"message":"An unexpected error occurred.","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}`

Any other suggestions for me to try?

many thanks!

**Reply by rijkvanzanten** [reply]
Hard to tell without having a hands-on with your instance and db!

**Reply by pandora-otto** [reply]
I'm running it locally, as I'm testing things out. Can I post some screenshots? Or is there a better way to show you what you need to see?

ty

**Reply by rijkvanzanten** [reply]
If there's no private data, a quick database dump is most helpful. That way I can just try it out locally and see what goes wrong 🙂

**Reply by pandora-otto** [reply]
I've got a database dump created. Whats the best way to send plz?

**Reply by pandora-otto** [reply]
I looked at the table 'Article Translations', which is auto generated, and the field is 'languages_code' (with an s).

So this` /items/articles?fields=id,name,translations&deep={"translations.*":{"_filter":{"languages_code":{"_eq":"en-US"}}}}`

gives me this:
`{"data":[{"id":1,"name":"My first article","translations":[1,2]}]}`

Just to be clear, I'd like to get the fields 'title' and 'text' that I have created in  'Article Translations', for the English language. At the moment it gives me [1,2] :/

By the way, same result I get If I run it without the &deep... section.

ty

**Reply by rijkvanzanten** [reply]
> I've got a database dump created. Whats the best way to send plz?

You can upload it right here 👍🏻

**Reply by rijkvanzanten** [reply]
> Just to be clear, I'd like to get the fields 'title' and 'text' that I have created in 'Article Translations', for the English language. At the moment it gives me [1,2] :/

Use the `fields` parameter 👍🏻 https://docs.directus.io/reference/api/query/#fields

**Reply by pandora-otto** [reply]
here it is,
[directus_dump.sql.zip](https://github.com/directus/directus/files/6880516/directus_dump.sql.zip)
thanks again for trying this

**Reply by rijkvanzanten** [reply]
```
GET /items/articles
  ?fields=*.*
  &deep[translations][_filter][languages_code][_eq]=en-US
``` 

is what you're looking for 🙂 

(or if you prefer JSON in the query:

```
GET /items/articles
  ?fields=*.*
  &deep={ "translations": { "_filter": { "languages_code": { "_eq": "en-US" }}}}
```

**Reply by pandora-otto** [reply]
wow, that is amazing! it works perfect!

thank you!

**Reply by pandora-otto** [reply]
Trying to work out the 'slideshows', the same way. This time it's like 3 levels.
Slideshows -> Slides -> Translations

I've tried this:
`/items/slideshows?fields=id,name,page,slides.*,slides.*.image,slides.*.heading,slides.*.translations.test&deep[translations][_filter][languages_code][_eq]=de-DE`

but it gives me all translations of the field 'test'

Can you spot my mistake?

**Reply by rijkvanzanten** [reply]
Yeah! The `translations` field is a couple levels deeper, so you have to "follow" that path:

```
GET /items/slideshows
  ?fields=id,name,page,slides.slides_id.*.*
  &deep[slides][slides_id][translations][_filter][languages_code][_eq]=de-DE
```

(also cleaned up your fields selection a notch)

**Reply by pandora-otto** [reply]
Aha, I see how the &deep is constructed in this case, it follows the path as you said, it makes sense really.
I could see it was something like this but I was trying something like `&deep[*.*.translations]` so it didn't work.

So thankful for this @rijkvanzanten I thought Directus was very clever but it turns out it's genius!
I think the future is big for this project.
I'll definitely get involved and try to help the community as I understand the team is not very big.

big big thank you!
